### PR TITLE
[action] allow filtering by build configuration in update_code_sign_settings

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation and Context
Allow finer grain control to update build settings only at a configuration level

### Description
- Iterate over all of the build configuration separately instead of modifying all at a target level

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: fastlane/lib/fastlane/actions/update_code_signing_settings.rb [hunk 1]
```diff
@@ -16,84 +16,86 @@ def self.run(params)
...
+            # Since Xcode 8, this is no longer needed, you simply use PROVISIONING_PROFILE_SPECIFIER
+            if params[:profile_uuid]
+              set_build_setting(config, "PROVISIONING_PROFILE", params[:profile_uuid])
+              UI.important("Set Provisioning Profile UUID to: #{params[:profile_uuid]} for target: #{target.name} for build configuration: #{config.name}")
+            end
+            if params[:bundle_identifier]
+              set_build_setting(config, "PRODUCT_BUNDLE_IDENTIFIER", params[:bundle_identifier])
+              UI.important("Set Bundle identifier to: #{params[:bundle_identifier]} for target: #{target.name} for build configuration: #{config.name}")
+            end
+
```
Comment (snatchev) [inline fastlane/lib/fastlane/actions/update_code_signing_settings.rb:62]: It's a little hard to follow what's going on with a long block inside of nested loop.

Suggestion: Collect the appropriate build configurations. Then map over them with a function that a function that takes a `config` and `params` and `target`(?). The return values can be used to accumulate `changed_targets` and `changed_build_configurations`.

Another option would be separate the collecting of the build configs as a separate function.

File: fastlane/lib/fastlane/actions/update_code_signing_settings.rb [hunk 2]
```diff
@@ -16,84 +16,86 @@ def self.run(params)
...
 
         if changed_targets.empty?
           UI.important("None of the specified targets has been modified")
           UI.important("available targets:")
-          target_dictionary.each do |target|
-            UI.important("\t* #{target[:name]}")
+          project.targets.each do |target|
+            UI.important("\t* #{target.name}")
           end
         else
```
Comment (snatchev) [inline fastlane/lib/fastlane/actions/update_code_signing_settings.rb:76]: can we exit the method here instead of the `else`?

File: fastlane/spec/actions_specs/update_code_signing_settings_spec.rb [hunk 3]
```diff
@@ -0,0 +1,233 @@
+require "xcodeproj"
```
Comment (snatchev) [inline fastlane/spec/actions_specs/update_code_signing_settings_spec.rb:1]: This file is just a rename?
